### PR TITLE
Remove mass solve from stage_type value

### PR DESCRIPTION
**Pull request by ScottMacLachlan**
The current implementation of `stage_type == "value"` fails when solving a DAE and using a non-stiffly accurate BT.  The math(s) behind that implementation are based on ODEs, so this is really just a case that fell through the cracks.

Doing a little pencil-and-paper work, you can easily reformulate the update to work for ODEs and DAEs, based on the computed stage values, without the mass solve.  This should be faster than what we had before, so it seems like this should be the default way we do things when the BT is not stiffly accurate...

### CODE REVIEW

**Review by rckirby — COMMENTED** [review]
I think this seems reasonable (you do catch the shortcut failing due to noninvertible Butcher matrix and fall back), but please respond to my comment, perhaps revealing a vague paranoia but perhaps seeing a possible corner case?

File: tests/test_dae.py [hunk 1]
```diff
@@ -42,3 +44,49 @@ def test_foo():
 
     assert not is_ode(F, up)
     assert is_ode(F + inner(Dt(p), w) * dx, up)
+
+
+@pytest.mark.parametrize('tableau', [RadauIIA, GaussLegendre])
```
[the remark this answers is not in the document]
Comment (pbrubeck) [reply tests/test_dae.py:53]: This test is now giving failures for stage-value + GaussLegendre (a non-stiffly accurate BT) in https://github.com/firedrakeproject/Irksome/pull/204 after rejiggering the `Dt` expansion.
